### PR TITLE
allow comments and whitespace before shader_type declaration

### LINKED ISSUE #16032: Comment on first line of shader program causes an inaccurate/non-intuitive error message
If I place a comment at the top of a shader program, it generates an unrelated/inaccurate "Assignment to constant expression" error message later in the program, e.g. on a line where I assign a value to COLOR. Here's a very simple code snippet to reproduce the issue:

```
// My test shader
shader_type canvas_item;

void fragment( )
{
	COLOR = vec4(1.0,1.0,1.0,1.0);
}
```

Delete the first line, and everything works fine. With the first line there, the error message is displayed on the line that assigns to COLOR. I think comments should be allowed before shader_type is defined, but if the core developers think otherwise, a more accurate error message would also be acceptable. Currently the error message is confusing.

**Godot version:**
V3.0, custom build (currently based on Commit #44cce2b of Jan 24, 2018, but the issue has been there for awhile).

### PR DESCRIPTION
**Pull request by brainsick**
Needs review.

I re-define a GETCHAR macro in this file.  Should I use another name?  Both uses are local to their respective functions.

Fixes #16032.

### CODE REVIEW

File: servers/visual/shader_language.cpp [hunk 1]
```diff
@@ -3990,13 +3990,56 @@ Error ShaderLanguage::_parse_shader(const Map<StringName, FunctionInfo> &p_funct
...
+				case '\t':
+				case '\r':
+				case '\n': {
+					idx++;
+				} break; // switch
+				default:
+					return idx;
+			}
+		}
+	}
```
Comment (reduz) [inline servers/visual/shader_language.cpp]: seems fine, but please undefine GETCHAR at the end

**Review by reduz — CHANGES_REQUESTED** [review]

**Comment by mhilbrunner** [comment]
@brainsick Can you make the requested change?

**Comment by akien-mga** [comment]
Merged manually as 9423f23ff with the required `#undef GETCHAR`. Also added one for the previous `GETCHAR` definition.

**Comment by akien-mga** [comment]
Now that Godot 3.1 has been released, we only plan to cherry-pick critical bug fixes to the 3.0 branch, unless there is very strong support in favor of a specific non-critical issue. Removing cherrypick label for 3.0.

**Pull request closed without merging**